NUnitTestProject1/Task3.cs

```diff
@@ -0,0 +1,141 @@
+
+        public char separator = ',';
+
+        string path = @"..\..\..\";
+
+        [TestCaseSource(nameof(PersonDictopnary))]
+        public void VerifyDictopnaryValue(List<Dictionary<string, string>> expectedDictionaryList, string fileName)
+        {
+            List<string> list = ReadFromFile(path + fileName);
+
+            Assert.AreEqual(expectedDictionaryList, FormatListToListDictionary(list));
+        }
+
+        private static IEnumerable<TestCaseData> PersonDictopnary()
+        {
+            #region Test_1 - 2 parameters
+
+            Dictionary<string, string> dictionary_1_1 = new Dictionary<string, string>
+            {
+                ["name"] = "Sergey",
+                ["surname"] = "Ivanov"
+            };
+
+            Dictionary<string, string> dictionary_1_2 = new Dictionary<string, string>
+            {
+                ["name"] = "Andrey",
+                ["surname"] = "Petrov"
+            };
+
+            List<Dictionary<string, string>> dictionaryList_1 = new List<Dictionary<string, string>> { dictionary_1_1, dictionary_1_2};
+
+            string fileName_1 = "PersonsDictionary_1.csv";
+
+            yield return new TestCaseData(dictionaryList_1, fileName_1);
+
+            #endregion Test_1 - 2 parameters
+
+            #region Test_2 - empty strings
+
+            Dictionary<string, string> dictionary_2_1 = new Dictionary<string, string>
+            {
+                ["name"] = "Sergey",
+                ["surname"] = "",
+                ["account"] = "123456"
+            };
+
+            Dictionary<string, string> dictionary_2_2 = new Dictionary<string, string>
+            {
+                ["name"] = "",
+                ["surname"] = "Petrov",
+                ["account"] = "23444"
+            };
+
+            List<Dictionary<string, string>> dictionaryList_2 = new List<Dictionary<string, string>> { dictionary_2_1, dictionary_2_2 };
+
+            string fileName_2 = "PersonsDictionary_2.csv";
+
+            yield return new TestCaseData(dictionaryList_2, fileName_2);
+
+            #endregion Test_2 - 3 parameters
+        }
+
+        private List<string> ReadFromFile(string path)
+        {
+            List<string>  valueListFromFile = new List<string>();
+
+            using (StreamReader sr = new StreamReader(path, System.Text.Encoding.Default))
+            {
+                string line;
+                while ((line = sr.ReadLine()) != null)
+                {
+
+                    valueListFromFile.Add(line);
+                }
+            }
+
+            return valueListFromFile;
+        }
+
+        private List<Dictionary<string, string>> FormatListToListDictionary(List<string> list)
+        {
+            string[] keys = GetKeysFromList(ref list);
+
+            return GetDictionaryList(keys, list);
+        }
+
+        private string[] GetKeysFromList(ref List<string> list)
+        {
+            string[] keys = list.First().Split(new[] { separator });
```

Comment:
```suggestion
            string[] keys = list[0].Split(new[] { separator });
```
Why do you use so complicated ways?